soda-core/src/soda_core/common/data_source_impl.py

```diff
@@ -203,6 +204,12 @@ def test_connection_error_message(self) -> Optional[str]:
     def build_data_source(self) -> DataSource:
         return DataSource(name=self.name, type=self.type_name)
 
+    def qualify_dataset_name(self, dataset_identifier: DatasetIdentifier) -> str:
+        assert dataset_identifier.data_source_name == self.name
```

Comment:
asserting like this seems unexpected, shouldn't this be a custom exception with a meaningful message?

Reply:
Added an error message, but kept the AssertionException because it's a bug, not a user error.  Can change to SodaException if you think that is better

Reply:
There is a lot of debate whether exceptions should be used for anticipated user flows or not, but using them for bugs is imho always valid